### ISSUE TITLE
"nosql": "^3.0.3" has a bug 

### ISSUE THREAD

**Issue by kanian** [oauthinaction/oauth-in-action-code#26]
To anybody going through the exercises after node v10.0.0,

nosql v3.0.3, used in the code exercises,  has a bug in FileReader.prototype.open in index.js line 2388. The [callback in fs signature is no longer optional](https://nodejs.org/api/fs.html#fs_fs_close_fd_callback) since v7.0.0 and throws an error since node v10.0.0
However, newer versions of nosql, have a different API than the one used in the exercises code. A quick fix, just to follow the exercises: in the node_modules/nosql/index.js, at line 2388, change 
`fs.close(fd);` to
`fs.close(fd, err=>{console.log(err)});`

Cheers

**Comment by amchavan** [comment]
Thank you Kanian.

**Comment by kanian** [comment]
You are welcome @amchavan

**Comment by BobJackson** [comment]
if your node version is newer,for example v10.15.0, you can fix it like following: 
1. update nosql version,for example , "nosql": "^5.0.0";
2. replace nosql.one() -> nosql.find().make(function(filter){});

This is my code: 
`
protectedResource.js
`
````
    nosql.find().make( function(filter) {
        filter.where('access_token', '=', inToken);
        filter.callback(function (err, token) {
            if (token) {
                console.log("We found a matching token: %s", inToken);
            } else {
                console.log('No matching token was found.');
            }
            req.access_token = token;
            next();
        });
    });
````

**Comment by jricher** [comment]
Should be fixed by #44 .
